lib/puppet/provider/route53_record.rb

```diff
@@ -39,23 +47,48 @@ def exists?
   end
 
   def record_hash(action)
-    values = resource[:values] || @property_hash[:values]
+    values = resource[:values]
```

Comment:
I think resource[:values] would actually be set from the @property_hash if no value was passed in, so net effect the same.